### DISCUSSION TITLE
Floaty Shits Aren't Spawning

### DISCUSSION THREAD

**Discussion by THEkatinamicrowave** [CodenameCrew/CodenameEngine#446] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/c8a700c2-f143-4188-9f54-20d7093225cc)

cif1.xml:
```xml
<!DOCTYPE codename-engine-stage>
<stage zoom="0.85" name="cif1" folder="stages/cif1/" startCamPosX="640" startCamPosY="360">
	<sprite name="background" x="0" y="0" sprite="void" scroll="0.85"/>
	<sprite name="floatydebris" x="0" y="0" sprite="testdebris" scroll="0.85"/>

	<girlfriend x="870" y="90" />
	<dad x="500" y="110" />
	<boyfriend x="1040" y="110" />
</stage>
```
cif1.hx:
```haxe
//
import flixel.tweens.FlxTween;

var accel:Float = 0;
var add:Boolean = true;
var elapsedCounter:Float = 0;
var interval:Float = 0.5;

function newShittySquare()
{
	var goofyrand:Float = 5 + (Math.random() * 5);
	var square:FlxSprite;

	square = new FlxSprite();
	square.makeSolid(goofyrand, goofyrand, 0xFFFF0000);
	square.x = Math.random() * (FlxG.width - square.width);
	square.y = 500 + (Math.random() * (FlxG.height - square.height - 500));
	square.moves = true;
	square.velocity.x = 1 + (Math.random() * 19);
	square.velocity.y = 40 + (Math.random() * 120);
	square.angularVelocity = (Math.random() * 180) - 90;
	add(square);

	// FlxTween.tween(square, {alpha: 0, "scale.x": 0, "scale.y": 0}, 3 + (Math.random() * 7)).start();

	trace(square);
}

function create()
{
	floatydebris.moves = true;
}

function update(elapsed:Float)
{
	// DEBRIS HANDLER ---------------------------------------------

	accel = add ? (accel + 1) : (accel - 1);
	add = (accel >= 120) ? false : (accel <= -120) ? true : add;
	floatydebris.velocity.y = accel;

	// FLOATY SHIT HANDLER ----------------------------------------

	elapsedCounter += elapsed;
	if (elapsedCounter >= interval)
	{
		newShittySquare();
		elapsedCounter = 0;
		interval = (Math.random() < 0.75) ? 1.0 : 0.5;
	}
}
```
Folder:
![image](https://github.com/user-attachments/assets/c7f89e18-9c9e-4438-8663-262269bcff7a)
Therefore, seemingly no reason for there to be no floaty shits

**Comment by r6915ee** [comment]
Does `void.png` at least spawn?

**Reply by THEkatinamicrowave** [reply]
yes. so does testdebris. testdebris also does the testdebris thing

https://github.com/user-attachments/assets/28324a9e-ae14-4108-8d07-2c2ee961529f

**Reply by r6915ee** [reply]
I would probably use [FlxObject](https://api.haxeflixel.com/flixel/FlxObject.html) instead for the squares, unless Codename doesn't really allow that. The HaxeFlixel API also doesn't seem to have that method for an `FlxSprite`, so I'm not sure why it would be used here, unless I'm missing some context with the fact that Codename uses special versions of certain things.

**Reply by THEkatinamicrowave** [reply]
random but I decided to check the logs during this
I trace square when the function runs and I found that it is, in fact, making the square
it seems that the issue is it not appearing visually
any ideas? let me know

**Reply by r6915ee** [reply]
I'm guessing it might just be the random values pushing away the square. I'm not exactly sure, the code looks like it's doing what it's supposed to.

**Reply by THEkatinamicrowave** [reply]
good idea
I'll test getting rid of the random values and see if that works

**Reply by THEkatinamicrowave** [reply, marked as answer]
bro.............. the problem was that I was using add() instead of FlxG.state.add()............ 🤦🏻‍♂️ whatever it works now
![image](https://github.com/user-attachments/assets/faa6173c-ae5a-4c2c-8075-2e45478e3960)

https://github.com/user-attachments/assets/ad79d068-0306-4ba3-ab74-c85074f58e7b

**Reply by r6915ee** [reply]
Now that's something I wouldn't have expected. It's good that it works now, though.

**Reply by THEkatinamicrowave** [reply]
I guess the real issue now is... why wasn't it adding to the FlxState in the first place?

**Reply by r6915ee** [reply]
I have zero clue myself.

**Reply by THEkatinamicrowave** [reply]
"ohhhhhhh NEEEEEEEEEEEEEeeeeeeeeeex!!!"



<weed>: @NexIsDumb

**Reply by NexIsDumb** [reply]
ho what

**Reply by THEkatinamicrowave** [reply]
why was it not adding to FlxG.state by default
context: this is a stage HX
context 2: literally read the thread

**Reply by NexIsDumb** [reply]
> why was it not adding to FlxG.state by default
> 
> context: this is a stage HX
> 
> context 2: literally read the thread

honestly no idea?? check if the parent of the script was indeed playstate or not

**Reply by chronicsilly** [reply]
> "ohhhhhhh NEEEEEEEEEEEEEeeeeeeeeeex!!!"
> 
> : @NexIsDumb

Please change your name, or I will be suing for misinformation. Your name is clearly "KadePleaseHelpMe", yet here you are - asking for help from someone who clearly ISN'T called Kade. My lawyers will be in touch within the week.

**Reply by charlesisfeline** [reply]
> Your name is clearly "KadePleaseHelpMe", yet here you are

isnt he also called "Microkat"

**Reply by THEkatinamicrowave** [reply]
> > "ohhhhhhh NEEEEEEEEEEEEEeeeeeeeeeex!!!"
> > : @NexIsDumb
> 
> Please change your name, or I will be suing for misinformation. Your name is clearly "KadePleaseHelpMe", yet here you are - asking for help from someone who clearly ISN'T called Kade. My lawyers will be in touch within the week.

![theyfoundthemeth](https://github.com/user-attachments/assets/42dd5f64-e352-4c49-b93d-ca5e7f4134df)

**Reply by THEkatinamicrowave** [reply]
> > Your name is clearly "KadePleaseHelpMe", yet here you are
> 
> isnt he also called "Microkat"

yes that's my main online name

**Reply by r6915ee** [reply]
> Please change your name, or I will be suing for misinformation. Your name is clearly "KadePleaseHelpMe", yet here you are - asking for help from someone who clearly ISN'T called Kade. My lawyers will be in touch within the week.

me when my username contains the name of a good horror series with an F appended at the end:
